task/factory/default/README.md

```diff
@@ -0,0 +1,321 @@
+# Default Task Factory
+The default [task factory](https://github.com/Knotx/knotx-fragments/blob/master/task/factory/api#task-factory) loads a task definition from the configuration and creates a [task](https://github.com/Knotx/knotx-fragments/blob/master/task/api#task)
+based on it. It uses the [actions' library](https://github.com/Knotx/knotx-fragments/blob/master/action#task)
+to do operations on the fragment (such as fetching data from 3rd party APIs).
+
+## How does it work
+It registers extendable graph node factories (one of them use Actions), delegates node initialization 
+to them and joins all nodes with transitions.
+
+Factory implements the [task factory interface](https://github.com/Knotx/knotx-fragments/blob/master/task/factory/api#task-factory). 
+It is a java library (a JAR file) that can be loaded using [Java SPI](https://docs.oracle.com/javase/tutorial/ext/basics/spi.html). 
+
+## How to configure
+For all configuration fields and their defaults consult [DefaultTaskFactoryConfig](https://github.com/Knotx/knotx-fragments/blob/master/task/factory/default/docs/asciidoc/dataobjects.adoc#defaulttaskfactoryconfig).
+
+In general:
+- it defines supported [tasks (by name)](#tasks) containing graph configurations in `tasks`
+- it converts the graph configurations into tasks using [node factories](#node-factory) defined in `nodeFactories`
+
+### Tasks
+[Tasks](https://github.com/Knotx/knotx-fragments/tree/master/task/api#task) are configured in 
+the form of a dictionary (`taskName -> graph logic`):
+```hocon
+tasks {
+  # unique task name
+  myTask { 
+    # graph configuration
+  }
+}
+```
+
+A graph configuration starts with a root [node](#node) definition. Each node sets some logic to perform 
+over a fragment and outgoing edges (called [transitions](#transition)). So there are two sections:
+```hocon
+node {
+  # node options (fragment processing logic)
+}
+onTransitions {
+  # node outgoing edges
+}
+```
+
+The `node` section contains some node options (e.g action name), the `onTransitions` defines transitions
+and assigned next nodes.
+
+#### Node
+The `node` section provides details on how the [node](https://github.com/Knotx/knotx-fragments/tree/master/task/api#node) 
+is instantiated.
+
+The default nodes:
+- [action node](#action-node)
+- [subtasks node](#subtasks-node)
+
+are initialized by [node factories](#node-factory).
+
+##### Action node
+An *action node* declares an action name to execute. An action name points to the action configured
+in the [action node factory](#action-node-factory).
+
+###### Full syntax
+```hocon
+node {
+  factory = action
+  config {
+    action = action-name
+  }
+}
+# onTransitions { }
+```
+
+###### Short syntax
+```hocon
+action = action-name
+# onTransitions { }
+```
+
+Please note that all options despite of `node.config.action` are skipped.
+
+##### Subtasks node
+Subtasks node is a node containing a list of subtasks. It evaluates all of them sequentially. 
```

Comment:
That's quite hard to understand. Maybe a picture can explain it better?
Because when we have:
```
A -> B -> C
D -> E
F
```
then:
- A, B and C are evaluated sequentially
but
- A, D, F are evaluated in parallel

Reply:
https://github.com/Knotx/knotx-fragments/issues/156